policy-controller/k8s/index/src/server.rs

```diff
@@ -312,3 +312,150 @@ impl Server {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+
+    use linkerd_policy_controller_core::ClientAuthentication;
+    use linkerd_policy_controller_k8s_api::policy::server::{Port, ProxyProtocol};
+
+    use crate::authz::AuthzIndex;
+
+    use super::*;
+
+    fn mk_server(
+        ns: impl Into<String>,
+        name: impl Into<String>,
+        port: Port,
+    ) -> k8s::policy::Server {
+        k8s::policy::Server {
+            api_version: "v1alpha1".to_string(),
+            kind: "Server".to_string(),
+            metadata: k8s::ObjectMeta {
+                namespace: Some(ns.into()),
+                name: Some(name.into()),
+                labels: None,
+                ..Default::default()
+            },
+            spec: k8s::policy::ServerSpec {
+                port,
+                pod_selector: Default::default(),
+                proxy_protocol: None,
+            },
+        }
+    }
+    trait MkServer {
+        fn with_proxy_protocol(self, p: ProxyProtocol) -> Self;
+        fn with_srv_labels(
+            self,
+            labels: impl IntoIterator<Item = (&'static str, &'static str)>,
+        ) -> Self;
+    }
+
+    impl MkServer for k8s::policy::Server {
+        fn with_proxy_protocol(mut self, p: ProxyProtocol) -> Self {
+            self.spec.proxy_protocol = Some(p);
+            self
+        }
+
+        fn with_srv_labels(
+            mut self,
+            labels: impl IntoIterator<Item = (&'static str, &'static str)>,
+        ) -> Self {
+            self.metadata.labels = Some(
+                labels
+                    .into_iter()
+                    .map(|(k, v)| (k.to_string(), v.to_string()))
+                    .collect(),
+            );
+            self
+        }
+    }
```

Comment:
I wouldn't use a trait for this -- we're not actually using it to be generic.  better to have free-functions:
```suggestion
    fn with_proxy_protocol(mut src: k8s::policy::Server, p: ProxyProtocol) -> Self {
        srv.spec.proxy_protocol = Some(p);
        srv
    }

    fn with_srv_labels(
        mut srv: k8s::policy::Server,
        labels: impl IntoIterator<Item = (&'static str, &'static str)>,
    ) -> Self {
        srv.metadata.labels = Some(
            labels
                .into_iter()
                .map(|(k, v)| (k.to_string(), v.to_string()))
                .collect(),
        );
        self
    }
```

Reply:
Nice! Does look better, pushed an update.